src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToInterpolatedString
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertToInterpolatedString), Shared]
+    internal partial class ConvertToInterpolatedStringRefactoringProvider :
+        AbstractConvertToInterpolatedStringRefactoringProvider
+            <InterpolatedStringExpressionSyntax,
+             InterpolatedStringContentSyntax,
+             InterpolationSyntax,
+             InvocationExpressionSyntax,
+             ExpressionSyntax,
+             ArgumentSyntax,
+             LiteralExpressionSyntax>
+    {
+        protected override SeparatedSyntaxList<ArgumentSyntax>? GetArguments(InvocationExpressionSyntax invocation) =>
+            invocation?.ArgumentList?.Arguments;
+
+        protected override LiteralExpressionSyntax GetFirstArgument(SeparatedSyntaxList<ArgumentSyntax> arguments) =>
+            arguments[0]?.Expression as LiteralExpressionSyntax;
+
+        protected override InterpolatedStringExpressionSyntax GetInterpolatedString(string text) =>
+            ParseExpression("$" + text) as InterpolatedStringExpressionSyntax;
+
+        protected override LiteralExpressionSyntax GetLiteralExpressionSyntax(SeparatedSyntaxList<ArgumentSyntax> arguments) =>
+            arguments[0].Expression as LiteralExpressionSyntax;
+
+        protected override LiteralExpressionSyntax GetLiteralExpressionSyntax(InterpolationSyntax interpolation) =>
+            interpolation.Expression as LiteralExpressionSyntax;
+
+        protected override string GetText(LiteralExpressionSyntax literalExpression)
+        {
+            var stringToken = literalExpression.Token;
+            var text = stringToken.ToString();
+
+            if (stringToken.IsVerbatimStringLiteral())
+            {
+                // We need to escape braces as this is an ambiguous case in interpolated strings that is 
+                // not ambiguous in verbatim strings
+                text = text.Replace(@"'{'", @"'{{'").Replace(@"'}'", @"'}}'");
+            }
+
+            return text;
+        }
+
+        protected override bool IsStringLiteral(LiteralExpressionSyntax firstArgument) =>
+            firstArgument?.IsKind(SyntaxKind.StringLiteralExpression) == true;
+
+        protected override ExpressionSyntax GetExpression(ArgumentSyntax argument) =>
```

Comment:
You can get this functionality through ISyntaxFacts service.  negating the need for the subclasses to provide this functionality.